src/ServiceControl.Audit.Persistence.RavenDB/EmbeddedDatabase.cs

```diff
@@ -28,7 +28,7 @@ public sealed class EmbeddedDatabase : IDisposable
                 isStopping = true;
             });
 
-            // Installer scenarios do not use the host and do not have a lifetime
+            // Test scenarios do not always have a lifetime
```

Comment:
This is the only bit I'm not so happy about. We make this nullable because of tests. We shouldn't have to account for how tests are set up in the production code.

Reply:
Agreed. I was also thinking it is not ideal but on the other hand we have also other places where we have internal properties and stuff that is only used for testing

Reply:
With the latest changes to the PR, all production scenarios have a host available, and I was able to modify the usage in the tests to pass in an `ApplicationLifetime`.